### PR TITLE
chore: add switch and switch group

### LINKED ISSUE #25509: Add Switch component

### LINKED ISSUE #25510: Add Switch Group component

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #25509 
Fixes #25510

### CODE REVIEW

**Comment by jsartisan** [comment]
@KelvinOm Note:
1. There is no Group component for Switch in react-aria, so using the `Checkboxgroup` only. Reexporting it as SwitchGroup
2. The code for Switch is mostly the same except that it uses `useSwitch` instead of `useCheckbox`

File: app/client/packages/design-system/widgets/src/components/SwitchGroup/index.tsx [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export { CheckboxGroup as SwitchGroup } from "../CheckboxGroup";
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/SwitchGroup/index.tsx:1]: Let's create abstract component for this purpose. Something like `SelectionGroup`. For Checkbox, Radio and Switch. 

Also I think it worth to add story for SwitchGroup, People need to know how to use this component.

File: app/client/packages/design-system/widgets/src/components/Switch/index.styled.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  &[data-focused] [data-icon] {
+    box-shadow: 0 0 0 2px var(--color-bg), 0 0 0 4px var(--color-bd-focus);
+  }
+
+  /**
+ * ----------------------------------------------------------------------------
+ * ERROR ( INVALID )
+ *-----------------------------------------------------------------------------
+ */
+  &[data-invalid] [data-icon] {
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Switch/index.styled.tsx:75]: Why do we need this? The switch does not have such a condition

File: app/client/packages/design-system/headless/src/components/Switch/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  // it should be safe in this case since the checkbox is not expected to be added or removed from the group.
+  const context = useContext(CheckboxGroupContext) as CheckboxGroupContextType;
+  const isDisabled = isDisabledProp || context?.isDisabled;
+  const { hoverProps, isHovered } = useHover({ isDisabled });
+  const { inputProps } = context?.state
+    ? // eslint-disable-next-line react-hooks/rules-of-hooks
+      useCheckboxGroupItem(
+        {
+          ...props,
+          // Value is optional for standalone checkboxes, but required for CheckboxGroup items;
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/Switch/Switch.tsx:52]: This is not checkbox.

**Review by KelvinOm — APPROVED** [review]

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**